modules/apps/portal-workflow/portal-workflow-kaleo-service/src/main/java/com/liferay/portal/workflow/kaleo/internal/upgrade/v1_4_1/KaleoDefinitionVersionUpgradeProcess.java

```diff
@@ -186,6 +194,54 @@ protected void upgradeKaleoDefinitionVersion() throws Exception {
 				}
 			}
 
+			if (hasTable("KaleoDraftDefinition")) {
+				PreparedStatement preparedStatement3 =
+					connection.prepareStatement(
+						"select * from KaleoDraftDefinition where " +
+							"draftVersion != 1");
```

Comment:
In the dump, I noticed that one of the KaleoDraftDefinition entries has no corresponding entry in KaleoDefinition. Since this entry also has a draftVersion of 1, under these changes, the entry will be ignored. I have since added [a fix](https://github.com/jesseyeh-liferay/liferay-portal/pull/30/commits/0e85d2fc579cb1098abbb312c5847dd5fc7a6656) which checks for all missing entries and am currently running the tests here: https://github.com/jesseyeh-liferay/liferay-portal/pull/30